platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/DelegateServletFilter.java

```diff
@@ -0,0 +1,289 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.pax.web.jetty;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.CopyOnWriteArraySet;
+import javax.annotation.Nullable;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link DelegateServletFilter} is meant to detect any {@link Filter}s and doFilter through those
+ * before continuing on its {@link FilterChain}. If no {@link Filter}s are found, it becomes a
+ * pass-through filter.
+ */
+public class DelegateServletFilter implements Filter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
+
+  private final Set<String> keysOfInitializedFilters;
+
+  @Nullable
+  private FilterConfig filterConfig;
+
+  public DelegateServletFilter() {
+    keysOfInitializedFilters = new CopyOnWriteArraySet<>();
+  }
+
+  BundleContext getContext() {
+    final Bundle bundle = FrameworkUtil.getBundle(DelegateServletFilter.class);
+    if (bundle != null) {
+      return bundle.getBundleContext();
+    }
+    return null;
+  }
+
+  @Override
+  public void init(@Nullable FilterConfig filterConfig) {
+    this.filterConfig = filterConfig;
+    // need to (re)initialize every Filter after the DelegateServletFilter is initialized
+    keysOfInitializedFilters.clear();
+  }
+
+  @Override
+  public void destroy() {
+    // do nothing
+  }
+
+  @Override
+  public void doFilter(
+      ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    TreeSet<ServiceReference<Filter>> sortedFilterServiceReferences = null;
+    final BundleContext bundleContext = getContext();
+
+    if (bundleContext == null) {
+      throw new ServletException(
+          "Unable to get BundleContext. No servlet Filters can be applied. Blocking the request processing.");
+    }
+
+    try {
+      sortedFilterServiceReferences =
```

Comment:
❓ Why do we pull refs through the bundle context instead of using a reference list through blueprint?

Reply:
Thinking about that made me realize that I could probably trim this class down quite a bit... but to answer your question it's basically just left over as I copied this from a previous implementation